react/features/base/media/components/web/AudioTrack.js

```diff
@@ -0,0 +1,216 @@
+     * element, as we set all the properties manually.
+     *
+     * @inheritdoc
+     * @returns {boolean} - False is always returned to blackbox this component
+     * from React.
+     */
+    shouldComponentUpdate(nextProps: Props) {
+        const currentJitsiTrack = this.props.audioTrack && this.props.audioTrack.jitsiTrack;
+        const nextJitsiTrack = nextProps.audioTrack && nextProps.audioTrack.jitsiTrack;
+
+        if (currentJitsiTrack !== nextJitsiTrack) {
+            this._detachTrack(this.props.audioTrack);
+            this._attachTrack(nextProps.audioTrack);
+        }
+
+        if (this._ref) {
+            const currentVolume = this._ref.volume;
+            const nextVolume = nextProps.volume;
+
+            if (typeof nextVolume === 'number' && currentVolume !== nextVolume) {
+                this._ref.volume = nextVolume;
+            }
+
+            const currentMuted = this._ref.muted;
+            const nextMuted = nextProps.muted;
+
+            if (typeof nextMuted === 'boolean' && currentMuted !== nextVolume) {
+                this._ref.muted = nextMuted;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { autoPlay, id } = this.props;
+
+        return (
+            <audio
+                autoPlay = { autoPlay }
+                id = { id }
+                ref = { this._setRef } />
+        );
+    }
+
+    /**
+     * Calls into the passed in track to associate the track with the component's audio element.
+     *
+     * @param {Object} track - The redux representation of the {@code JitsiLocalTrack}.
+     * @private
+     * @returns {void}
+     */
+    _attachTrack(track) {
+        if (!track || !track.jitsiTrack) {
+            return;
+        }
+
+        track.jitsiTrack.attach(this._ref);
+    }
+
+    /**
+     * Removes the association to the component's audio element from the passed
+     * in redux representation of jitsi audio track.
+     *
+     * @param {Object} track -  The redux representation of the {@code JitsiLocalTrack}.
+     * @private
+     * @returns {void}
+     */
+    _detachTrack(track) {
+        if (this._ref && track && track.jitsiTrack) {
+            track.jitsiTrack.detach(this._ref);
+        }
+    }
+
+    _setRef: (?HTMLAudioElement) => void;
+
+    /**
+     * Sets the reference to the HTML audio element.
+     *
+     * @param {HTMLAudioElement} audioElement - The HTML audio element instance.
+     * @private
+     * @returns {void}
+     */
+    _setRef(audioElement: ?HTMLAudioElement) {
+        this._ref = audioElement;
+        const { onAudioElementReferenceChanged } = this.props;
+
+        if (this._ref && onAudioElementReferenceChanged) {
+            onAudioElementReferenceChanged({ volume: this._ref.volume });
```

Comment:
I was not sure if the reference never changes. Are you sure this is correct? At the end of the day it is managed by React internally...

Anyway even if it is called only initially this will be fine. We are using it to read the initial volume value in the Thumbnail in order to set the volume slider.

I admit the naming of the prop is not the best one. I had troubles coming up with a name here. If you have better ideas I will be more than happy to change it!

Reply:
I see. I think what confused me is the fact that we only use it for the volume but it has some pretty generic name. How about making it explicitly just for the volume and call it `onInitialVolumeSet` ?

Reply:
Fixed!